src/components/dashboard/dashboard.spec.ts

```diff
@@ -0,0 +1,93 @@
+import { Component } from '@angular/core';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { DashboardOptions } from './dashboard.component';
+import { DashboardModule } from './dashboard.module';
+
+
+@Component({
+    selector: 'app-ux-dashboard',
+    template: `<ux-dashboard [options]="options"></ux-dashboard>`
+})
+export class DashboardComponent {
+
+    options: DashboardOptions  = {
+        columns: 3,
+        padding: 10,
+        rowHeight: 300,
+        emptyRow: false,
+        minWidth: 187
+    };
+}
+
+describe('Dashboard', () => {
+    let component: DashboardComponent;
+    let fixture: ComponentFixture<DashboardComponent>;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [DashboardModule],
+            declarations: [DashboardComponent]
+        })
+            .compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(DashboardComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+    });
+
+    it('should initialise correctly', () => {
+        expect(component).toBeTruthy();
+    });
+
+    it('should change columns option if columns option adjusted', async() => {
+        expect(component.options.columns).toBe(3);
+
+        component.options.columns = 6;
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        expect(component.options.columns).toBe(6);
```

Comment:
I believe the original plunker proved that the options object did not revert back to the original value after being updated. The problem was the dashboard itself not updating, which is what these tests should aim to verify by inspecting the rendered DOM. This one for example might be done by checking the dashboard's aria-label.

Reply:
I agree, I don't think these test really prove much. This is why I decided to do the e2e tests. The problem I've got with this is that I need to inspect the widgets rather than the dashboard itself. I had a few goes at introducing widgets to the tests but I couldn't find a way of actually getting them to behave correctly.

Reply:
What was the problem?

Reply:
The widgets just stacked on top of each other. I tried just using an individual widget but still no css was taking effect. Maybe I was looking at this from the wrong angle and I should have been testing the widget-component. The widget gets all its style attributes via @HostBinding and from what I can see in plunker, it was getting the option changes, its just the dashboard wasn't rendering on change. This is why I think the e2e test is the way to go to prove this fix. Since there wasn't a dashboard testbed, is it worth keeping it just in case its required in the future?

Reply:
Actually on looking back at it again, the widgets were not getting the updated options until the window was resized causing the dashboard to re render.

Reply:
I think some CSS must be taking effect if they are stacking on top of each other, since that would require `position: absolute`. Bear in mind that a LOT of the styling is custom on that dashboard example, you can check by opening the plunker.

However, we don't really care what it looks like for unit tests. Things like the row height and min width should be programmatically applied to the respective elements, so should be possible to confirm the effects of all of these options regardless of other styling or animation state.
![image](https://user-images.githubusercontent.com/26114101/75244058-f130ca80-57c2-11ea-8a14-1916b9e47774.png)

Reply:
> 
> 
> Actually on looking back at it again, the widgets were not getting the updated options until the window was resized causing the dashboard to re render.

Yes, as noted below you are not triggering change detection because you are not changing the object reference of `options`.